docker/synapse.Dockerfile

```diff
@@ -10,13 +10,8 @@ RUN apt-get -qq update && apt-get -qq install -y \
         apt-utils ${PYTHON_VERSION} ${PYTHON_VERSION}-dev ${PYTHON_VERSION}-venv \
         python3-pip eatmydata redis-server
 
-RUN ${PYTHON_VERSION} -m pip install --cache-dir /pip-cache poetry==1.1.14 && \
-    rm -rf /pip-cache
-# Instead of using `--no-cache-dir`, we specify a `--cache-dir` and delete it
-# afterwards. This works around a bug where `msgpack` gets installed to
-# `site-packages`, where Python can't find it, instead of `dist-packages`, when
-# using `setuptools>=48.0.0` and the `--no-cache-dir` option.
-# https://github.com/matrix-org/sytest/issues/1269
+RUN ${PYTHON_VERSION} -m pip install -q --upgrade pip
```

Comment:
Should we be _pinning_ the version of pip here?

Reply:
I can't convince myself that we should pin or not pin pip either way.
The dependencies of poetry are equally free to update underneath us. So perhaps we should always upgrade pip, in case one of them starts relying on some new pip behaviour?

Adding a comment sounds like a good idea.